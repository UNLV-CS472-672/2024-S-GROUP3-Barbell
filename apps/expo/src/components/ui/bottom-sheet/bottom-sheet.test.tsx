import React from 'react'

import { render } from '@testing-library/react-native'

import CustomBottomSheet from './bottom-sheet'

/**
 * @bug https://github.com/software-mansion/react-native-reanimated/issues/1555
 * but its fixed it seems, leave for reference
 */
jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'))
jest.mock('@gorhom/bottom-sheet', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(({ children }) => children),
  useBottomSheet: jest.fn(() => ({ close: jest.fn() })),
}))

describe('CustomBottomSheet', () => {
  it('renders correctly', () => {
    const screen = render(<CustomBottomSheet title="Test Title" />)

    expect(screen.getByTestId('button-test')).toBeTruthy()
    expect(screen.getByTestId('button-test-2')).toBeTruthy()
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
