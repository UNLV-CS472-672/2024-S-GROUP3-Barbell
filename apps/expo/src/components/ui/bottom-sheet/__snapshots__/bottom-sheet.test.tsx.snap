// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomBottomSheet renders correctly 1`] = `
<View
  className="mt-7"
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="active:opacity-30 bg-[#717171] w-11/12 py-3 rounded-lg mb-5"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="button-test"
  >
    <Text
      className="text-center  text-white text-3xl font-semibold font-koulen"
      testID="button-text"
    >
      Start Saved Workout
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="active:opacity-30 bg-[#717171] w-11/12 py-3 rounded-lg"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="button-test-2"
  >
    <Text
      className="text-center  text-white text-3xl font-semibold font-koulen"
      testID="button-text"
    >
      Create New Workout
    </Text>
  </View>
</View>
`;
